A72-open-telemetry-tracing.md

```diff
@@ -0,0 +1,560 @@
+Note that the in-binary migration and cross-process migration can be done in parallel.
+
+The shim package that bridges two libraries works as follows, considering the
+following migration scenarios example: 
+
+```agsl
+|-- Application - Configured OpenCensus ------------------------------- |
+|--  gRPC -> Using OpenCensus to generate Trace A  -------------------- |
+|--  Application -> Using OpenCensus to generate a sub Trace B--------- |
+```
+
+The application may use a bridge package in the outermost layer first:
+```agsl
+|-- Application - Configured Otel w/ OpenCensus Shim ------------------- |
+|--  gRPC -> Using OpenCensus to generate Trace A  --------------------- |
+|--  Application -> Using OpenCensus to generate a sub Trace B---------- |
+```
+
+Then the application changes the instrumentation to OpenTelemetry:
+```agsl
+|-- Application - Configured Otel w/ OpenCensus Shim ---------------------- |
+|--  gRPC -> Using OpenCensus to generate Trace A  -------------------------|
+|--  Application -> Using Otel to generate a sub Trace B------------------- |
+```
+
+Finally, they switch to grpc-open-telemetry and finish the migration.
+```agsl
+|-- Application - Configured Otel standalone ----------------------------- |
+|--  gRPC -> Using Otel to generate Trace A  ----------------------------- |
+|--  Application -> Using Otel to generate a sub Trace B------------------ |
+```
+### Tracing Information
+gRPC is generating similar tracing information for OpenTelemetry compared with OpenCensus,
+but due to API differences between those two libraries, the
+trace information is represented slightly differently.
+In the new OpenTelemetry plugin, the client will add `Event`s (name:
+`Outbound message sent` and `Inbound message read`) with corresponding attributes,
+mapped from OpenCensus `MessageEvent` fields:
+
+| OpenCensus Trace Message Event Fields | OpenTelemetry Trace Event Attribute Key |
+|---------------------------------------|-----------------------------------------|
+| Type                                  | `message.event.type`                    |
+| Message Id                            | `message.message.id`                    |
+| Uncompressed message size             | `message.event.size.uncompressed`       |
+| Compressed message size               | `message.event.size.compressed`         |
+
+OpenCensus span annotation description maps to OpenTelemetry event name, and
+annotation attributes keys are mapped to event attributes keys:
+
+| OpenCensus Trace Annotation Attribute Key | OpenTelemetry Trace Event Attribute Key |
+|-------------------------------------------|-----------------------------------------|
+| `type`                                    | `message.event.type`                    |
+| `id`                                      | `message.message.id`                    |
+
+And OpenTelemetry no longer has span end options as OpenCensus does.
```

Comment:
Does that mean we cannot get RPC status from traces?

Reply:
The status will be recorded in a separate API `.setStatus()` for OTel Java. In Census Java the status was in span end option.
However, I noticed Otel C++ and Census C++ always has `.setStatus()` API.
So I removed this line and rephrased it in the `Tracing Information` section in the proposal. I think it is more clear. Thanks!